sale_coupon_order_pending/models/coupon_coupon.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2023 Tecnativa - David Vidal
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+
+
+class SaleCoupon(models.Model):
+    _inherit = "coupon.coupon"
+
+    can_be_applied_to_order = fields.Boolean(
+        compute="_compute_can_be_applied_to_order",
+        compute_sudo=True,
+    )
+
+    @api.depends_context("active_id")
+    def _compute_can_be_applied_to_order(self):
+        self.can_be_applied_to_order = False
+        if self.env.context.get("active_model", "") != "sale.order":
+            return
+        order = self.env["sale.order"].browse(self.env.context.get("active_id"))
+        for coupon in self:
+            if not coupon._check_coupon_code(
+                order.date_order.date(), coupon.partner_id.id
+            ):
+                self.can_be_applied_to_order = True
```

Comment:
This can even be more simplified
```suggestion
            self.can_be_applied_to_order = not coupon._check_coupon_code(
                order.date_order.date(), order.partner_id.id, order=order
            )
```

Reply:
Thanks for the suggestions, I was using the method wrong. This also solved test issues.